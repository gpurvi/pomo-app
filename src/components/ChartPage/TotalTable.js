import React from 'react';
import LoadingCircle from "../common/LoadingCircle";

const TotalTable = ({loading, totalSessions, totalHours, averageSessions, averageHours}) => (
    <div className='container mt-6 pt-3'>
        <div className='position-relative'>
            <LoadingCircle height='h-100' loading={loading}/>
            <div className='row'>
                <div className='col-6 text-center'>
                    {`Total sessions: ${totalSessions}`}
                </div>
                <div className='col-6 text-center'>
                    {`Total hours: ${totalHours}`}
                </div>
            </div>

            <div className='row'>
                <div className='col-12'>
                    <hr className='mt-1 mb-1'/>
                </div>

            </div>
            <div className='row'>
                <div className='col-6 text-center'>
                    {`Average sessions per day: ${averageSessions}`}
                </div>
                <div className='col-6 text-center'>
                    {`Average hours per day: ${averageHours}`}
                </div>
            </div>
        </div>
    </div>
);

export default TotalTable;