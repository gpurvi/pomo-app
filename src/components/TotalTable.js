import React from 'react';

const TotalTable = (props) => (
    <div>
        <div>
            Total sessions: {props.totalSessions}
            Total hours: {props.totalHours}
        </div>
        <hr/>
        <div>
            Average sessions per day: {props.aveSessions}
            Average hours per day: {props.aveHours}
        </div>
    </div>
);

export default TotalTable;