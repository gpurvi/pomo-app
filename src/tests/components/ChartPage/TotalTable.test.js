import React from 'react';
import {shallow} from 'enzyme';
import TotalTable from '../../../components/ChartPage/TotalTable';

it('should render ChartButtons', () => {
    const wrapper = shallow(<TotalTable
        totalSessions={142}
        totalHours={13}
        averageSessions={1}
        averageHours={5}
    />);
    expect(wrapper).toMatchSnapshot();
});
