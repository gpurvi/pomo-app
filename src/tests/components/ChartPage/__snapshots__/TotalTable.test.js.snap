// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ChartButtons 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TotalTable
    averageHours={5}
    averageSessions={1}
    totalHours={13}
    totalSessions={142}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          Total sessions: 142 Total hours: 13
</div>,
        <hr />,
        <div>
          Average sessions per day: 1 Average hours per day: 5
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Total sessions: 142 Total hours: 13",
        },
        "ref": null,
        "rendered": "Total sessions: 142 Total hours: 13",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Average sessions per day: 1 Average hours per day: 5",
        },
        "ref": null,
        "rendered": "Average sessions per day: 1 Average hours per day: 5",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Total sessions: 142 Total hours: 13
</div>,
          <hr />,
          <div>
            Average sessions per day: 1 Average hours per day: 5
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Total sessions: 142 Total hours: 13",
          },
          "ref": null,
          "rendered": "Total sessions: 142 Total hours: 13",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Average sessions per day: 1 Average hours per day: 5",
          },
          "ref": null,
          "rendered": "Average sessions per day: 1 Average hours per day: 5",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
